x-pack/plugins/security/server/session_management/session.ts

```diff
@@ -358,24 +370,54 @@ export class Session {
   }
 
   /**
-   * Clears session value for the specified request.
-   * @param request Request instance to clear session value for.
+   * Invalidates sessions that match the specified filter.
+   * @param request Request instance initiated invalidation.
+   * @param filter Filter that narrows down the list of the sessions that should be invalidated.
    */
-  async clear(request: KibanaRequest) {
+  async invalidate(request: KibanaRequest, filter: InvalidateSessionsFilter) {
+    // We don't require request to have the associated session, but nevertheless we still want to
+    // log the SID if session is available.
     const sessionCookieValue = await this.options.sessionCookie.get(request);
-    if (!sessionCookieValue) {
-      return;
-    }
+    const sessionLogger = this.getLoggerForSID(sessionCookieValue?.sid);
 
-    const sessionLogger = this.getLoggerForSID(sessionCookieValue.sid);
-    sessionLogger.debug('Invalidating session.');
+    let invalidateIndexValueFilter;
+    if (filter.match === 'current') {
+      if (!sessionCookieValue) {
+        return;
+      }
 
-    await Promise.all([
-      this.options.sessionCookie.clear(request),
-      this.options.sessionIndex.clear(sessionCookieValue.sid),
-    ]);
+      sessionLogger.debug('Invalidating current session.');
+      await this.options.sessionCookie.clear(request);
```

Comment:
**question** what benefits to we get by invalidating the session cookie?

Clearing the session cookie is very tightly coupled to the way that Hapi decides to handle this, so I worry about unintended side effects with hapi upgrades. If there's an error clearing the session cookie, then we won't perform the actual session invalidation, which is arguably the more important aspect.

If we really should clear the session cookie here, then I think we should do one of two things:
1) Catch any errors thrown by `sessionCookie.clear`, so that this operation may continue
2) Clear the session cookie _after_ we've invalidated the requested server-side sessions.

Reply:
> question what benefits to we get by invalidating the session cookie?

No real benefits to be honest, cleared it just because we could :slightly_smiling_face: 

> Clearing the session cookie is very tightly coupled to the way that Hapi decides to handle this, so I worry about unintended side effects with hapi upgrades. If there's an error clearing the session cookie, then we won't perform the actual session invalidation, which is arguably the more important aspect.

That's an interesting point, I didn't think about that. I'm not sure if the risk is high, but I agree that it still exists. Let me just drop this line then to eliminate the potential problem completely.